enterprise/app/api_keys/api_keys.tsx

```diff
@@ -576,3 +573,73 @@ function newFormState<T extends ApiKeyFields>(request: T): FormState<T> {
     request,
   };
 }
+
+interface ApiKeyFieldProps {
+  value: string;
+}
+
+interface ApiKeyFieldState {
+  value: string;
+  isCopied: boolean;
+  hideValue: boolean;
+  displayValue: string;
+}
+
+const ApiKeyFieldDefaultState: ApiKeyFieldState = {
+  value: "",
+  isCopied: false,
+  hideValue: true,
+  displayValue: "************",
+};
+
+class ApiKeyField extends React.Component<ApiKeyFieldProps, ApiKeyFieldState> {
+  state = ApiKeyFieldDefaultState;
+
+  componentDidMount() {
+    this.setState({ value: this.props.value });
+  }
+
+  // onClick handler function for the copy button
+  private handleCopyClick() {
+    const copyTextToClipboard = async (text: string) => {
+      if ("clipboard" in navigator) {
+        return await navigator.clipboard.writeText(text);
+      } else {
+        return document.execCommand("copy", true, text);
+      }
+    };
```

Comment:
We have an existing util, `app/util/clipboard.ts` for this

Reply:
let me try using it 🤔